decorator/go/comporession-decorator.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import "fmt"
+
+// Yeni davranışların tanımlandığı sınıftır.
+// UML diyagramındaki ConcreteDecorator sınıfına denk gelmektedir.
+type compressionDecorator struct {
```

Comment:
Diğer örneklere baktığımda `BaseDataSourceDecorator` var ekstradan, burada bir tek o eksik gibi ama çok da problem değil gibi geldi bana.

Reply:
abstract class'in go da bir karşılığı olmadığı icin sadece methodları  interface kullanarak implementasyonu sağlanıyor. Ondan dolayı eklemeyi ben de gerekli görmedim. 😄